fabric-rendering-fluids-v1/src/main/java/net/fabricmc/fabric/api/client/render/fluid/v1/SimpleFluidRenderHandler.java

```diff
@@ -99,42 +134,82 @@ public SimpleFluidRenderHandler(Identifier stillTexture, Identifier flowingTextu
 	 * transparent blocks}.
 	 */
 	public SimpleFluidRenderHandler(Identifier stillTexture, Identifier flowingTexture, Identifier overlayTexture) {
-		this(stillTexture, flowingTexture, overlayTexture, -1);
+		this(stillTexture, flowingTexture, overlayTexture, -1, null);
 	}
 
 	/**
-	 * Creates a fluid render handler without an overlay texture and a custom,
-	 * fixed tint.
+	 * Creates a fluid render handler without an overlay texture,
+	 * with a custom fixed tint, with custom fog.
+	 *
+	 * @param stillTexture The texture for still fluid.
+	 * @param flowingTexture The texture for flowing/falling fluid.
+	 * @param tint The fluid color RGB. Alpha is ignored.
+	 * @param fogHandler The fluid fog rendering handler.
+	 */
+	public SimpleFluidRenderHandler(Identifier stillTexture, Identifier flowingTexture, int tint, @Nullable FluidFogHandler fogHandler) {
+		this(stillTexture, flowingTexture, null, tint, fogHandler);
+	}
+
+	/**
+	 * Creates a fluid render handler without an overlay texture,
+	 * with a custom fixed tint, with no fog.
 	 *
 	 * @param stillTexture The texture for still fluid.
 	 * @param flowingTexture The texture for flowing/falling fluid.
 	 * @param tint The fluid color RGB. Alpha is ignored.
 	 */
 	public SimpleFluidRenderHandler(Identifier stillTexture, Identifier flowingTexture, int tint) {
-		this(stillTexture, flowingTexture, null, tint);
+		this(stillTexture, flowingTexture, null, tint, null);
+	}
+
+	/**
+	 * Creates a fluid render handler without an overlay texture,
+	 * with no tint, with custom fog.
+	 *
+	 * @param stillTexture The texture for still fluid.
+	 * @param flowingTexture The texture for flowing/falling fluid.
+	 * @param fogHandler The fluid fog rendering handler.
+	 */
+	public SimpleFluidRenderHandler(Identifier stillTexture, Identifier flowingTexture, @Nullable FluidFogHandler fogHandler) {
```

Comment:
I think this one clashes with the `(Identifier, Identifier, Identifier)` one when the last parameter is null. I don't think these extra constructors are necessary though? I would say keep the most generic one only, since a custom fog handler is quite niche.

Reply:
I've removed the constructors with the fog handler, and added a new fogHandler method to set the handler.
The method returns `this`, so you can use like `FabricBlockSettings.of(Material.METAL).strength(4.0f)`.

`new SimpleFluidRenderHandler(...).fogHandler(handler)`